url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/Treatment.java

```diff
@@ -0,0 +1,121 @@
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.util.HashSet;
+import java.util.Set;
+
+import RQLibrary.EncodingSymbol;
+
+public class Treatment extends Thread {
+    final static int size_int = 4;
+    byte[][] matrice;
+    long[] temps = new long[1000];
+    private final Set<EncodingSymbol> receivedPackets = new HashSet<EncodingSymbol>();
+    int RTT;
+    boolean start;
+    boolean boucle;
+    int nb;
+    int nbRecv;
+    int ESI;
+    
+    Treatment(int rtt) {
+	
+	RTT = rtt;
+	
+	for (int i = 0; i < temps.length; i++) {
+	    temps[i] = 0;
+	}
+	nb = 0;
+	nbRecv = 0;
+	start = true;
+	boucle = true;
+	ESI = 1;
+	start();
+	
+    }
+    
+    int getESI() {
+	return this.ESI;
+    }
+    
+    int getNb() {
+	return this.nb;
+    }
+    
+    int getNbRecv() {
+	return this.nbRecv;
+    }
+    
+    void setEndBoucle() {
+	boucle = false;
+    }
+    
+    public void run() {
+	// look and compare all time if > RTT pop it
+	while (boucle) {
+	    for (int i = 0; i < temps.length; i++) {
+		
+		if ( temps[i] != 0 ) {
+		    // if ( System.currentTimeMillis() - temps[i] > 500 ) {
+		    if ( System.currentTimeMillis() - temps[i] > this.RTT ) {
+			nbRecv++;
+			temps[i] = 0;
+			receivedPackets.add((EncodingSymbol) treatPacket(matrice[i]));
+		    }
+		}
+	    }
+	    try {
+		Thread.sleep(10);
+	    } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)